### PR TITLE
Update threat model to include rollback attack

### PR DESCRIPTION
**Pull request by priteshbandi**
Update threat model to include rollback attack

### CODE REVIEW

File: threatmodels/notation-threatmodel.md [hunk 1]
```diff
@@ -87,3 +87,4 @@ The certificates trusted by the verifier are stored in Notation trust store in t
...
 | Compromised Notation dependencies                              | Tampering              | Mitigated | High     | Notation       | The dependencies that built into Notation binary was compromised, this may lead to arbitrary code being executed | Notation keeps dependencies up-to-date and adds new dependency after careful consideration and only if it's absolutely required. Always use static build instead of dynamic linking |
+| Rollback Attack                              | Tampering              | Mitigated | High     | Notation       | Attacker can exploit a compromised repository to return old vulnerable artifact | Signature publisher can employ short signature expiration periods (and periodically re-sign artifacts) or revoke outdated vulnerable artifacts |
```
Comment (yizha1) [inline threatmodels/notation-threatmodel.md]: We have an actor of Signer. Maybe use Signer instead signature producer is better.
```suggestion
| Rollback Attack                              | Tampering              | Mitigated | High     | Notation       | Attacker can exploit a compromised repository to return outdated vulnerable artifacts | Signer can employ short signature expiration periods (and periodically re-sign artifacts) or revoke outdated vulnerable artifacts |
```

**Review by yizha1 — COMMENTED** [review]
Thanks @priteshbandi left some comments

**Review by gokarnm — APPROVED** [review]
LGTM!

**Review by yizha1 — APPROVED** [review]
LGTM

**Review by rgnote — APPROVED** [review]
LGTM

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**